awsrds/rds_db_instance.go

```diff
@@ -122,6 +123,42 @@ func (r *RDSDBInstance) DescribeSnapshots(DBInstanceID string) ([]*DBSnapshotDet
 	return dbSnapshotDetails, nil
 }
 
+func (r *RDSDBInstance) DeleteSnapshots(keepForDays int) error {
+	r.logger.Info("delete-old-rds-snapshots", lager.Data{"keep_for_days": keepForDays})
+
+	deleteBefore := time.Now().Add(-1 * time.Duration(keepForDays) * 24 * time.Hour)
+
+	snapshotsToDelete := []string{}
+
+	err := r.rdssvc.DescribeDBSnapshotsPages(
+		&rds.DescribeDBSnapshotsInput{},
+		func(page *rds.DescribeDBSnapshotsOutput, lastPage bool) bool {
+			for _, snapshot := range page.DBSnapshots {
+				if snapshot.SnapshotCreateTime.Before(deleteBefore) {
```

Comment:
If I understand correctly, this will delete all snapshots in the AWS account that are older than N days. I think we should add additional constraints to this so that it only deletes snapshots that are related to instances managed by the broker.

We set a "Broker Name" tag on the instances which looks to get copied to the snapshots. We should filter for snapshots with this tag. (In our deployments, "Broker Name" is set to the `DEPLOY_ENV`)

Reply:
I added filtering for the `Broker Name` tag and we only delete manual snapshots.